From 495431f926eb72d58ababd678c20f2ad45f24bbe Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Mon, 12 Jun 2023 10:55:19 -0700
Subject: [PATCH 02/11] UPSTREAM: bug#64034: [PATCH] libparted: link
 libparted-fs-resize.so to libuuid

uuid_generate() is used in this library
Hence, add UUID_LIBS to libparted_fs_resize_la_LIBADD to avoid

"ld.lld: error: undefined reference due to --no-allow-shlib-undefined: uuid_generate"

This is uncovered by lld linker linking fatresize package
due to this linker defaulting to --no-allow-shlib-undefined.

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Signed-off-by: Brian C. Lane <bcl@redhat.com>

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 libparted/fs/Makefile.am | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libparted/fs/Makefile.am b/libparted/fs/Makefile.am
index 41a60d9..7fa8d14 100644
--- a/libparted/fs/Makefile.am
+++ b/libparted/fs/Makefile.am
@@ -75,6 +75,7 @@ libparted_fs_resize_la_LDFLAGS =		\
 EXTRA_DIST += fsresize.sym
 libparted_fs_resize_la_DEPENDENCIES = $(sym_file)
 
+libparted_fs_resize_la_LIBADD  = $(UUID_LIBS)
 libparted_fs_resize_la_SOURCES = \
   r/filesys.c			\
   r/fat/bootsector.c		\
-- 
2.52.0

